consensus-types/blocks/proofs.go

```diff
@@ -0,0 +1,99 @@
+package blocks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state/stateutil"
+	field_params "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v5/encoding/ssz"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+	"go.opencensus.io/trace"
+)
+
+func ComputeFieldRootsForBlockBody(ctx context.Context, blockBody *BeaconBlockBody) ([][]byte, error) {
+	ctx, span := trace.StartSpan(ctx, "ComputeFieldRootsForBlockBody")
+	defer span.End()
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
+	if blockBody == nil {
+		return nil, errNilBlockBody
+	}
+	var fieldRoots [][]byte
+	switch blockBody.version {
+	case version.Phase0:
+		fieldRoots = make([][]byte, 11)
+	case version.Altair:
+		fieldRoots = make([][]byte, 9)
+	case version.Bellatrix:
+		fieldRoots = make([][]byte, 10)
+	case version.Capella:
+		fieldRoots = make([][]byte, 11)
+	case version.Deneb:
+		fieldRoots = make([][]byte, 12)
+	case version.Electra:
+		fieldRoots = make([][]byte, 12)
```

Comment:
How did you come up with these numbers? They should be equal to the number of fields in the spec:
- 8 for Phase0: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#beaconblockbody
- 9 for Altair: https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#beaconblockbody
- etc

Reply:
> How did you come up with these numbers? They should be equal to the number of fields in the spec:
> 
> * 8 for Phase0: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#beaconblockbody
> * 9 for Altair: https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#beaconblockbody
> * etc

My bad I just counted them manually from the structs ``BeaconBlockBodyAltair`` ``BeaconBlockBodyCapella`` and so on.